src/php/Command/FormatCommand.php

```diff
@@ -4,23 +4,91 @@
 
 namespace Phel\Command;
 
+use Phel\Command\Shared\CommandIoInterface;
 use Phel\Formatter\FormatterInterface;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
 
 final class FormatCommand
 {
     public const COMMAND_NAME = 'fmt';
 
+    private const PHEL_EXTENSION = 'phel';
+
     private FormatterInterface $formatter;
+    private CommandIoInterface $io;
 
-    public function __construct(FormatterInterface $formatter)
-    {
+    /** @var array<string, bool> */
+    private array $alreadyTriedToFormatFilePath = [];
+
+    /** @var list<string> */
+    private array $successfulFormattedFilePaths = [];
+
+    public function __construct(
+        FormatterInterface $formatter,
+        CommandIoInterface $io
+    ) {
         $this->formatter = $formatter;
+        $this->io = $io;
+    }
+
+    /**
+     * @param list<string> $paths
+     */
+    public function run(array $paths): void
+    {
+        foreach (array_unique($paths) as $path) {
```

Comment:
Not fully of what I meant. Maybe it was a bad example. The next example will also run the same file twice:
```
phel fmt src/ src/test.phel
```
It would be better to have a function that takes `$paths` and returns a list of all files. On this list you can do `array_unique` and maybe also sort it, so the execution order is always the same.

Reply:
Ah, got it. Ok. I am on it